frontend/components/tree/js/tree.service.ts

```diff
@@ -1027,51 +1034,70 @@ export class TreeService {
 		this.ViewerService.clearHighlights();
 		this.DocsService.closeDocs();
 
-		if (this.currentSelectedNodes.length) {
-			while (this.currentSelectedNodes.length) {
-				const currentNode = this.currentSelectedNodes.pop();
-				currentNode.selected = this.SELECTION_STATES.unselected;
-				this.setSelectionOnParentNodes(currentNode, this.SELECTION_STATES.unselected);
+		const visitedParents = [];
+		const visitedChildren = [];
+
+		while (this.currentSelectedNodes.length) {
+			const currentNode = this.currentSelectedNodes.pop();
+			currentNode.selected = this.SELECTION_STATES.unselected;
```

Comment:
this line can be moved within the if statement in 1045. If you've visited this before surely you've flagged this as unselected.